### PR TITLE
Add scrypto dev container target

### PR DESCRIPTION
**Pull request by azizi-a**
## Summary
- Add new target in Docker file: **scrypto-dev-container**
- Define old Docker file target: **scrypto-builder**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -50,6 +50,13 @@ WORKDIR /app
 
 RUN cargo install --path ./radix-clis
 
+FROM base-image as dev-container
+RUN apt install -y bash-completion powerline
+RUN echo 'powerline-daemon -q; POWERLINE_BASH_CONTINUATION=1; POWERLINE_BASH_SELECT=1; . /usr/share/powerline/bindings/bash/powerline.sh; . /etc/bash_completion' >> ~/.bashrc
+COPY --from=builder /app/target/release/scrypto /usr/local/bin/scrypto
+COPY --from=builder /app/target/release/resim /usr/local/bin/resim
```
[the remark this answers is not in the document]
Comment (azizi-a) [reply Dockerfile:61]: Added the first two, but `scrypto_bindgen` was not found in `/app/target/release/scrypto_bindgen` when building the image.

File: Dockerfile [hunk 2]
```diff
@@ -50,6 +50,13 @@ WORKDIR /app
...
+RUN rustup target add wasm32-unknown-unknown
```
[the remark this answers is not in the document]
Comment (azizi-a) [reply Dockerfile:64]: [The documentation](https://code.visualstudio.com/docs/devcontainers/create-dev-container#_dockerfile) shows no examples with an `ENTRYPOINT`

File: Dockerfile [hunk 3]
```diff
@@ -50,7 +50,23 @@ WORKDIR /app
...
+RUN curl -sS https://starship.rs/install.sh | sh -s -- -y
+RUN echo 'eval "$(starship init bash)"\n . /etc/bash_completion' >> /root/.bashrc
+
+COPY --from=builder /app/target/release/scrypto /usr/local/bin/scrypto
+COPY --from=builder /app/target/release/resim /usr/local/bin/resim
+COPY --from=builder /app/target/release/rtmc /usr/local/bin/rtmc
+COPY --from=builder /app/target/release/rtmd /usr/local/bin/rtmd
+COPY --from=builder /app/target/release/scrypto-bindgen /usr/local/bin/scrypto-bindgen
+RUN rustup target add wasm32-unknown-unknown
+RUN rustup component add rustfmt
```
Comment (dhedey) [inline Dockerfile:67]: Ohh nice! We should probably add clippy too?

**Review by dhedey — APPROVED** [review]

File: Dockerfile (hunk 3 above)
Comment (azizi-a) [reply Dockerfile:67]: Done 👍🏽

**Pull request merged**